lineapy/utils/analytics/usage_tracking.py

```diff
@@ -33,6 +35,32 @@ def _py_version():
     )
 
 
+@lru_cache(maxsize=1)
+def _runtime():
+    if get_ipython() is None:
+        runtime = "non-ipython"
+    else:
+        envars: str = ";".join(list(os.environ))
+        if "DATABRICKS_" in envars:
+            runtime = "ipython-databricks"
+        elif "COLAB_" in envars:
+            runtime = "ipython-colab"
+        elif "DEEPNOTE_" in envars:
+            runtime = "ipython-deepnote"
+        elif "BINDER_" in envars:
+            runtime = "ipython-binder"
+        else:
+            runtime = "ipython"
+
+    # Support optional custom tag which can be used to flag
+    # and discard certain events such as those from dev work
+    tag = os.environ.get("LINEAPY_RUNTIME_TAG")
```

Comment:
good idea!

Reply:
Yeah, this implements @mingjerli's idea!